### PR TITLE
fix: proxy typing + copyrights

### PR DESCRIPTION
**Pull request by axel7083**
This PR includes

- This PR will __enforce__ at compilation the methods declared in the StudioAPI are under the format 
```ts
(...args: unknown[]): Promise<unknown>
```
- Move the `playground.ts` to the manager directory

### CODE REVIEW

**Review by lstocchi — APPROVED** [review]
Needs to be rebased.

**Review by feloy — APPROVED** [review]

**Comment by axel7083** [comment]
Fix for https://github.com/projectatomic/studio-extension/issues/95 added

File: packages/backend/src/managers/catalogManager.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import type { Catalog } from '@shared/src/models/ICatalog';
```
Comment (benoitf) [inline packages/backend/src/managers/catalogManager.ts:1]: FYI it's missing copyright headers

File: packages/backend/src/managers/catalogManager.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import path from 'node:path';
+import fs from 'node:fs';
+import defaultCatalog from '../ai.json';
+import type { Category } from '@shared/src/models/ICategory';
+import type { Recipe } from '@shared/src/models/IRecipe';
+import type { ModelInfo } from '@shared/src/models/IModelInfo';
+
+export class CatalogManager {
+  private catalog: Catalog;
```
Comment (benoitf) [inline packages/backend/src/managers/catalogManager.ts:10]: to make it really private you could use #

```suggestion
  #catalog: Catalog;
```

Comment (axel7083) [reply packages/backend/src/managers/catalogManager.ts:10]: Yeah but for readability, user not knowing what # mean, will not think it is priovate, can we use both ?
```
private #catalog: Catalog
```

Comment (benoitf) [reply packages/backend/src/managers/catalogManager.ts:10]: no you can't

Comment (axel7083) [reply packages/backend/src/managers/catalogManager.ts:31]: Gonna keep private for now as we are not yet using the #

**Pull request merged**